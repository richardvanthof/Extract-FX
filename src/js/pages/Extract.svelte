<script>
    import DropDown from "../components/Form/DropDown.svelte";
    import Switch from "../components/Form/Switch.svelte";
    import ExcludeModal from "../components/Form/ExcludeModal.svelte";
    import { setContext } from 'svelte';
    import { sourceTrack, destination, exclusions, exclusionOptions, isExclusionModalOpen } from '../global-vars/extract';
    import { trackTotal } from '../global-vars/shared';
    import { generateNumberedOptions } from '../lib/helpers';
    import {debugMode} from '../../../secrets';

    const setDestination = (type) => $destination = type; // Updates the destination store
    const trackOptions = generateNumberedOptions($trackTotal, 'VIDEO');
    const currentSource = $derived($sourceTrack);

    setContext('exclusionOptions', $exclusionOptions);
</script>

<form class="grid-container">
    <div class="grid-column">
        <div class="group">
            <label for="source-track">Source track</label>
            <DropDown options={trackOptions} value={currentSource} store={sourceTrack}  />
        </div>
        <div class="group">
            <label for="destination">Destination</label>
            <Switch 
                selected={$destination} 
                callback={setDestination} 
                options={['file', 'track']} 
                id="destination"
            />
        </div>
    </div>
    <div class="grid-column">
        <ExcludeModal {exclusions} options={exclusionOptions} open={isExclusionModalOpen}/>
    </div>
</form>

{#if debugMode}
<details open>
    <summary>Debug</summary>
    <h3>Extract: global vars.</h3>
    <ul>
        <li>Source track: {$sourceTrack}</li>
        <li>Destination: {$destination}</li>
        <li>Exclusions: {JSON.stringify($exclusions)}</li>
        <li>Exclusion options: {$exclusionOptions}</li>
        <li>Exclusion modal open: {$isExclusionModalOpen}</li>
    </ul>
</details>

{/if}
