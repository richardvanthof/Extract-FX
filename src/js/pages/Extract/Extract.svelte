<script lang='ts'>
    import DropDown from "@/js/components/Form/DropDown/DropDown.svelte";
    import Switch from "../../components/Form/Switch/Switch.svelte";
    import ExcludeModal from "../../components/Form/ExclusionModal/ExcludeModal.svelte";
    import { setContext } from 'svelte';
    import { globals } from '../../global-vars/globals.svelte';
    import {generateNumberedOptions} from '@/js/helpers/helpers.svelte';

    let {exclusionOptions, trackTotal } = globals;

    const trackOptions = $derived(generateNumberedOptions(trackTotal, 'VIDEO'));

    setContext('exclusionOptions', exclusionOptions);

    const handleDestinationUpdate = (selection: string) => {
      if(selection === 'file'|| selection ==='sequence') {
        globals.extract.destination = selection
      };
    };
</script>

<form data-testid="extract-form" class="grid-container">
    <div class="grid-column">
        <div class="group">
            <label for="source-track">Source track</label>
            <DropDown options={trackOptions} bind:value={globals.extract.sourceTrack}/>
        </div>
        <div class="group">
            <label for="destination">Destination</label>
            <Switch 
                value={globals.extract.destination}
                options={['File', 'Sequence']} 
                callback={handleDestinationUpdate}
            />
        </div>
    </div>
    <div class="grid-column">
        <ExcludeModal 
            bind:exclusions={globals.extract.exclusions} 
            bind:open={globals.isExclusionModalOpen} 
        />
    </div>
</form>