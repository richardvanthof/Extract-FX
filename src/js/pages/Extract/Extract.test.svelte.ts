import { render } from '@testing-library/svelte';
import '@testing-library/jest-dom';
import { describe, expect, it, afterEach, vi } from 'vitest';
import Extract from '@/js/pages/Extract/Extract.svelte';

describe('Extract.svelte', () => {
  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('Renders correct layout', () => {
    const { container } = render(Extract);
    expect(container).toMatchSnapshot();
  })
});
