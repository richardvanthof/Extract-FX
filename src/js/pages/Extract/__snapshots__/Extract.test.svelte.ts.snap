// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extract.svelte > Renders correct layout 1`] = `
<div>
  <form
    class="grid-container"
    data-testid="extract-form"
  >
    <div
      class="grid-column"
    >
      <div
        class="group"
      >
        <label
          for="source-track"
        >
          Source track
        </label>
         
        <!---->
        <select
          class="svelte-14mby5s"
          data-testid="dropdown-bind"
          id="sourceTrack"
        >
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="1"
          >
            VIDEO 1 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="2"
          >
            VIDEO 2 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="3"
          >
            VIDEO 3 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="4"
          >
            VIDEO 4 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="5"
          >
            VIDEO 5 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="6"
          >
            VIDEO 6 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="7"
          >
            VIDEO 7 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="8"
          >
            VIDEO 8 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="9"
          >
            VIDEO 9 
          </option>
          <option
            class="svelte-14mby5s"
            data-testid="dropdown-option"
            value="10"
          >
            VIDEO 10 
          </option>
          
        </select>
        
        <!---->
      </div>
       
      <div
        class="group"
      >
        <label
          for="destination"
        >
          Destination
        </label>
         
        <ul
          class="switch-input svelte-2z4jz9"
        >
          <button
            class="switch-option active svelte-2z4jz9"
          >
            File
          </button>
          <button
            class="switch-option  svelte-2z4jz9"
          >
            Sequence
          </button>
          
        </ul>
        <!---->
      </div>
    </div>
     
    <div
      class="grid-column"
    >
      <details
        class="wrapper svelte-1ojv71p"
        open=""
      >
        <summary
          class="exclusion-header svelte-1ojv71p"
          id="exclusion-header"
        >
          Exclude effects 
        </summary>
         
        <div
          class="exclusions svelte-1ojv71p"
        >
          <div
            class="exclusions-controls svelte-1ojv71p"
          >
            <button
              class="exclusion-toolbar-button exclusion-control svelte-1ojv71p"
              id="add-exclusion-btn"
              title="add"
            >
              + Add Exclusion
            </button>
             
            <button
              class="exclusion-control svelte-1ojv71p"
              id="remove-all-exclusions-btn"
              title="clear"
            >
              Clear all
            </button>
          </div>
           
          <div
            class="exclusions-container svelte-1ojv71p"
          >
            <div
              class="exclusions-placeholder svelte-1ojv71p"
            >
              <p
                class="caption"
              >
                Press the +-button to add an exclusion.
              </p>
            </div>
            <!---->
          </div>
        </div>
      </details>
      <!---->
    </div>
  </form>
  
</div>
`;
